js/frontend.js

```diff
@@ -1,41 +1,154 @@
+  async function postData(url, data = {}) {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    return response.json();
+  }
+
+  postData(serverUrl, bookEntry)
+    .then(data => console.log(data))
+    .catch(error => console.error(error));
+
+  console.log(bookEntry);
+  formBookTitle.value = "";
+  formWordGoals.value = "";
+  document.querySelector("#books").innerHTML = "";
+  setTimeout(() => {
+    getBooks(serverUrl);
+  }, 0300);
+});
+
+async function getBooks(url) {
+  try {
+    const response = await fetch(url, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json"
+      }
+    });
+    const data = await response.json();
+    const bookEntries = Array.from(data);
+    if (bookEntries.length > 0) {
+      createElements(bookEntries);
+      bookSubmissions.style.visibility = "visible";
     }
- 
-    async function postData(url, data = {}){
-        const response = await fetch(url, {
-            method: 'POST', 
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(data)
-        })
-        return response.json()
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function daysDifference(date) {
+  const oneDay = 24 * 60 * 60 * 1000; // numerical value for one day to be used to count difference for deadlines
+  const dueDate = new Date(date);
+  const currentDate = new Date(Date.now());
+  const daysLeft = Math.round(Math.abs((dueDate - currentDate) / oneDay));
+  return daysLeft;
+}
+
+function createElements(data) {
+  data.forEach(book => {
+    const card = document.createElement("div");
+    const cardBody = document.createElement("div");
+    const wordsToComplete = document.createElement("div");
+    const dateDue = document.createElement("div");
+    const row = document.createElement("div");
+    const deleteIcon = document.createElement("a");
+    let cardTitle = document.createElement("h1");
+    const wordsCompleted = document.createElement("div");
+    const daysUntilDue = document.createElement("div");
+    daysUntilDue.innerHTML = `<h5>Days left: ${daysDifference(book.deadline)} </h5>`;
+
+
+    let currentWords = 0;
+
+    wordsCompleted.innerHTML = `<h5>Words completed: ${currentWords}</h5> <a href="#">Update words completed</a>`;
+    wordsCompleted.classList.add("col-md-4");
+    row.appendChild(wordsCompleted);
+
+    deleteIcon.innerHTML = "Delete entry";
+    deleteIcon.classList.add("delete-icon");
+    deleteIcon.classList.add("p-2");
+    deleteIcon.style.color = "red";
+    deleteIcon.style.visibility = "hidden";
+
+    dateDue.classList.add("col-md-4");
+    dateDue.classList.add("text-center");
+    dateDue.classList.add("deadlineDate");
+    dateDue.innerHTML = `<h5>Deadline: ${book.deadline}</h5>`;
+
+    setTimeout(() => {
+      const deadlineDate = Array.from(
```

Comment:
No need to use Array.from here, `querySelectorAll` returns a collection that already has a `forEach` function on it.